buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/publish/PublishingPlugin.groovy

```diff
@@ -155,6 +178,13 @@ class PublishingPlugin implements Plugin<Project> {
         }
     }
 
+    private static Set<String> changedPaths(File workDir) {
+        return 'git diff --name-only --submodule=diff HEAD@{0} HEAD@{1}'
```

Comment:
I would prefer to see if git can be run programmatically. But it is probably fine if you think this task will be invoked only on machines that has git available.

Reply:
It diffs head and head - 1. I copied this from the checkChanged tasks. I think this is primarily intended for Prow, as it might give misleading results locally.